sycl-fusion/passes/cleanup/Cleanup.cpp

```diff
@@ -0,0 +1,164 @@
+//==---------------------------- Cleanup.cpp -------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Cleanup.h"
+
+#include <algorithm>
+#include <utility>
+
+#include <llvm/ADT/SmallVector.h>
+#include <llvm/IR/Attributes.h>
+#include <llvm/IR/DerivedTypes.h>
+#include <llvm/IR/LLVMContext.h>
+#include <llvm/IR/Metadata.h>
+
+#include "Kernel.h"
+#include "kernel-info/SYCLKernelInfo.h"
+
+using namespace llvm;
+
+static FunctionType *createMaskedFunctionType(const BitVector &Mask,
+                                              const FunctionType *FTy) {
+  SmallVector<Type *> Params;
+  std::transform(Mask.set_bits_begin(), Mask.set_bits_end(),
+                 std::back_inserter(Params),
+                 [&](unsigned I) { return FTy->getParamType(I); });
+  return FunctionType::get(FTy->getReturnType(), Params, false);
+}
+
+static void copyAttributesFrom(const BitVector &Mask, Function *NF,
+                               const Function *F) {
+  // Copies attributes, calling convention and other relevant data.
+  NF->copyAttributesFrom(F);
+  // Drop masked-out attributes.
+  SmallVector<AttributeSet> Attributes;
+  const llvm::AttributeList PAL = NF->getAttributes();
```

Comment:
nit: `llvm::` namespace is used inconsistently: there is `using namespace llvm;`, but in some places `llvm::` is still used. For example, below on line 44 `AttributeList` is referenced without `llvm::`. This comment also applies to other files